src/main/java/com/blackducksoftware/integration/hub/alert/datasource/relation/manager/HubUserManager.java

```diff
@@ -0,0 +1,240 @@
+        deleteConfig(objectTransformer.stringToLong(id));
+    }
+
+    public void deleteConfig(final Long id) {
+        if (doesConfigExist(id)) {
+            hubUsersRepository.delete(id);
+            hubUserFrequenciesRepository.delete(id);
+            if (hubUserEmailRepository.exists(id)) {
+                hubUserEmailRepository.delete(id);
+            }
+            if (hubUserHipChatRepository.exists(id)) {
+                hubUserHipChatRepository.delete(id);
+            }
+            if (hubUserSlackRepository.exists(id)) {
+                hubUserSlackRepository.delete(id);
+            }
+            final List<HubUserProjectVersionsRelation> hubUserProjectVersions = hubUserProjectVersionsRepository.findByUserConfigId(id);
+            for (final HubUserProjectVersionsRelation hubUserProjectVersion : hubUserProjectVersions) {
+                final HubUserProjectVersionsRelationPK key = new HubUserProjectVersionsRelationPK();
+                key.userConfigId = hubUserProjectVersion.getUserConfigId();
+                key.projectName = hubUserProjectVersion.getProjectName();
+                key.projectVersionName = hubUserProjectVersion.getProjectVersionName();
+                hubUserProjectVersionsRepository.delete(key);
+            }
+        } else {
+            logger.warn("Attempted to delete Hub User config with id {}, but it did not exist.", id);
+        }
+    }
+
+    private void saveFrequency(final Long id, final String frequency) {
+        final HubUserFrequenciesRelation hubUserFrequency = hubUserFrequenciesRepository.findOne(id);
+        if (hubUserFrequency != null) {
+            final String storedFrequency = hubUserFrequency.getFrequency();
+            if (storedFrequency.equals(frequency)) {
+                // No need to update the frequency.
+                return;
+            }
+        }
+        hubUserFrequenciesRepository.save(new HubUserFrequenciesRelation(id, frequency));
+    }
+
+    private <R extends DatabaseChannelRelation> void saveChannelId(final Long id, final String channelId, final JpaRepository<R, Long> repository, final Class<R> clazz) {
+        if (doesChannelConfigNeedUpdate(id, channelId, repository)) {
+            if (channelId != null) {
+                R channelRelation;
+                try {
+                    channelRelation = clazz.newInstance();
+                } catch (InstantiationException | IllegalAccessException e) {
+                    logger.error("Unable to save Hub user channel config (hubUserId: {}, channelConfigId: {}) because {} could not be instantiated: {}.", id, channelId, clazz.getSimpleName(), e.getLocalizedMessage());
+                    return;
+                }
+                channelRelation.setUserConfigId(id);
+                channelRelation.setChannelConfigId(objectTransformer.stringToLong(channelId));
+                repository.save(channelRelation);
+            } else {
+                repository.delete(id);
```

Comment:
We want to delete the mapping from user to channel id if the channel id is null. A mapping should only be in the db if it is configured.